packages/eas-build-job/src/generic.ts

```diff
@@ -22,9 +23,15 @@ export namespace Generic {
   export type Job = z.infer<typeof JobZ>;
   export const JobZ = z.object({
     projectArchive: ArchiveSourceSchemaZ,
+    // TODO: remove this once we pass all of the jobs to Turtle as objects
     customBuildConfig: z.object({
       path: z.string(),
     }),
+    customJob: z
```

Comment:
Let's make it `steps: z.array(...).optional()` so we don't have a (custom) job inside a (generic) job?